awx_collection/plugins/module_utils/tower_api.py

```diff
@@ -530,7 +534,11 @@ def create_if_needed(self, existing_item, new_item, endpoint, on_create=None, it
         # Process any associations with this item
         if associations is not None:
             for association_type in associations:
-                self.modify_associations(response, associations[association_type])
+                if response['json']['related'][association_type]:
+                    self.modify_associations(response['json']['related'][association_type], associations[association_type])
+                else:
+                    self.fail_json(msg="Association type of {0} is not valid for {1}".format(association_type, item_type))
```

Comment:
I don't think this is as helpful as it looks, because you have to consider the context someone has who sees it. Right now `item_type` is not consistently passed by the modules (sometimes it is, sometimes it isn't). That leaves the message referring to "unknown".

And anyway, we can basically assume that we'll add tests for any associations (even if you use the generator, these things are manually selected). If you are adding a relationship, getting this message is less helpful than just getting a traceback because it erases the context of how you got there. I have another PR where I'm trying to print tracebacks when they are turned from the module, and it works much better. There's very little hunting involved if it will just behave the way that python projects normally behave.